src/Kafka/Consumer.hs

```diff
@@ -62,28 +62,38 @@ newConsumer :: MonadIO m
             => ConsumerProperties
             -> Subscription
             -> m (Either KafkaError KafkaConsumer)
-newConsumer cp (Subscription ts tp) = liftIO $ do
-  kc@(KafkaConf kc' ct) <- newConsumerConf cp
+newConsumer props (Subscription ts tp) = liftIO $ do
+  let cp = setCallback (rebalanceCallback (\_ _ -> return ())) <> props
+  kc@(KafkaConf kc' qref ct) <- newConsumerConf cp
   tp' <- topicConf (TopicProps $ M.toList tp)
   _   <- setDefaultTopicConf kc tp'
-  rdk <- bimap KafkaError Kafka <$> newRdKafkaT RdKafkaConsumer kc'
-  case flip KafkaConsumer kc <$> rdk of
-    Left err -> return $ Left err
-    Right kafka -> do
-      forM_ (cpLogLevel cp) (setConsumerLogLevel kafka)
-      sub <- subscribe kafka ts
-      case sub of
-        Nothing  -> runEventLoop kafka ct (Just $ Timeout 100) >> return (Right kafka)
+  rdk <- newRdKafkaT RdKafkaConsumer kc'
+  case rdk of
```

Comment:
I really need to start using `ExceptT` or something like it for these sorts of things and to avoid nested `case` statements that are there because of `IO (Either a b)`